nixos/modules/services/home-automation/home-assistant.nix

```diff
@@ -251,6 +434,14 @@ in {
       '' else ''
         rm -f "${cfg.configDir}/ui-lovelace.yaml"
         ln -s ${lovelaceConfigFile} "${cfg.configDir}/ui-lovelace.yaml"
+      '') + (if (cfg.customLovelaceModules != []) then ''
+        ln -s ${customLovelaceModulesDir} "${cfg.configDir}/www/nixos-lovelace-modules"
+      '' else ''
+        rm "${cfg.configDir}/www/nixos-lovelace-modules"
+      '') + (if cfg.customComponents != [] then ''
+        ln -sf "${customComponentsDir}" "${cfg.configDir}/custom_components"
```

Comment:
With `pkgs.buildEnv` we are the only one who can manage the `custom_components` directory. A patch would be needed to create a custom one managed by nixos. I don't plan on writing one, but anyone feel free to propose one.